### PR TITLE
[Feature] Introducing Searching for ISO 3166 and Countries Ids

### LINKED ISSUE #226: API Query Issue ISO 3166
This is the Issue I meant...
https://corona.lmao.ninja/countries/au

It shows Austria instead of Australia... the ISO3/ISO2 of Austria is AT or AUT while Australia AUS or AU

Same thing with Philippines... there should be a response for PHL...
https://corona.lmao.ninja/countries/phl ---> no response.

Another issue is... 
https://corona.lmao.ninja/countries/ar should be argentina not Denmark

These are the issues I'm facing...

### PR DESCRIPTION
**Pull request by buster95**
This PR add the possiblity to do searches with ISO and Id
ex.
/countries/{iso2}
/countries/{iso3}
/countries/{country id}
/countries/{country name}

resolve #226
closed #228

### CODE REVIEW

File: utils/country_utils.js [hunk 1]
```diff
@@ -311,8 +310,19 @@ const getCountryData = function (countryName) {
...
+    }
+    return false;
+}
+
 module.exports = {
     getCountryCode,
     getCountryName,
-    getCountryData
+    getCountryData,
+    isCountryException,
```
Comment (Nekiono) [inline utils/country_utils.js]: can we not have trailing commas

File: server.js [hunk 2]
```diff
@@ -89,7 +102,7 @@ app.get("/countries/:country", async function (req, res) {
     return;
   }
   // adding status code 404 not found and sending response
-  res.status(404).send({ message: "Country not found or Has no cases" });
+  res.status(404).send({ message: "Country not found or doesn't have cases" });
```
Comment (Nekiono) [inline server.js]: doesn't have **any** cases maybe? this doesn't sound very fluent

File: server.js [hunk 3]
```diff
@@ -70,17 +70,30 @@ app.get("/historical/:country", async function (req, res) {
...
-
-  const standardizedCountryName = countryMap.standardizeCountryName(req.params.country.toLowerCase());
-  let country = countries.find(e => {
-    // see if strict was even a parameter
-    if (req.query.strict) {
-      return req.query.strict.toLowerCase() == 'true' ? e.country.toLowerCase() === standardizedCountryName : e.country.toLowerCase().includes(standardizedCountryName)
-    }
-    else {
-      return e.country.toLowerCase().includes(standardizedCountryName);
+  const { search } = req.params, isText = isNaN(search);
```
Comment (Nekiono) [inline server.js]: should be split into two assignments for readability

File: server.js [hunk 4]
```diff
@@ -70,17 +70,30 @@ app.get("/historical/:country", async function (req, res) {
...
+
+  let country = countries.find(ctry => {
```
Comment (Nekiono) [inline server.js]: is constant

File: server.js [hunk 5]
```diff
@@ -70,17 +70,30 @@ app.get("/historical/:country", async function (req, res) {
   res.send(countryData);
 });
 
-app.get("/countries/:country", async function (req, res) {
+app.get("/countries/:search", async (req, res) => {
```
Comment (Nekiono) [inline server.js]: query, not search

File: README.md [hunk 6]
```diff
@@ -29,6 +29,9 @@ NovelCovid API is recomended by Postman [here](https://covid-19-apis.postman.com
 |  https://corona.lmao.ninja/countries | Returns data of all countries that has COVID-19 |
 |  https://corona.lmao.ninja/countries?sort={parameter} | Returns data of each country sorted by the parameter |
 |  https://corona.lmao.ninja/countries/{country-name} | Returns data of a specific country. If an exact name match is desired pass ?strict=true in the query string |
+|  https://corona.lmao.ninja/countries/{country-Id} | Returns the data of a country if it is equal to its country id  |
```
Comment (Nekiono) [inline README.md]: if iso isn't capitalized, then id shouldn't be as well

Comment (Nikhil-Wagh) [reply README.md]:
```suggestion
|  https://corona.lmao.ninja/countries/{country-id} | Returns the data of a country if it is equal to its country id  |
```

File: server.js (hunk 5 above)
Comment (buster95) [reply server.js]: search is for all, ISO, Id and country name, What do you suggest?

File: server.js (hunk 4 above)
Comment (buster95) [reply server.js]: sorry fixed

File: server.js (hunk 2 above)
Comment (buster95) [reply server.js]: sorry fixed

File: server.js (hunk 3 above)
Comment (buster95) [reply server.js]: sorry fixed

File: utils/country_utils.js (hunk 1 above)
Comment (buster95) [reply utils/country_utils.js]: sorry fixed

File: README.md (hunk 6 above)
Comment (buster95) [reply README.md]: Sorry was fixed

**Comment by EliteDaMyth** [comment]
maybe add a thing to search for country ID cuz some people might not be using full country names rn for search so might break their projects :P

**Comment by buster95** [comment]
@EliteDaMyth This PR supports old features adding new ones, them projects will not break

**Comment by EliteDaMyth** [comment]
Seems good, I am seeking a review from Ethan and if all's good ill merge!

File: server.js (hunk 5 above)
Comment (ebwinters) [reply server.js]: she's talking about naming, query is better

Comment (buster95) [reply server.js]: fixed

**Comment by ebwinters** [comment]
@buster95  pls stop pushing I need to push my changes

**Comment by ebwinters** [comment]
@buster95 pls allow edits from maintainers on your fork https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork

**Pull request merged**